src/Core/Framework/DependencyInjection/seo.xml

```diff
@@ -159,5 +159,9 @@
             <argument type="service" id="slugify"/>
             <argument type="tagged" tag="shopware.seo_url.twig.extension"/>
         </service>
+
+        <service id="shopware.seo_url.twig_parser" class="Shopware\Core\Framework\Adapter\Twig\TwigVariableParser">
```

Comment:
this should not be needed when you change the constructor of the service directly, and also please mark the "regular" `TwigVariableParser` as deprecated too and refactor all core usages to the new factory